src/app/dataset-view/additional-components/dataset-settings-component/tabs/webhooks/create-subscription-modal/create-subscription-modal.component.ts

```diff
@@ -0,0 +1,101 @@
+import { AbstractControl, FormBuilder, Validators } from "@angular/forms";
+import { NgbActiveModal } from "@ng-bootstrap/ng-bootstrap";
+import AppValues from "src/app/common/values/app.values";
+import {
+    DatasetBasicsFragment,
+    WebhookSubscriptionInput,
+    WebhookSubscriptionStatus,
+} from "src/app/api/kamu.graphql.interface";
+import { WebhooksService } from "src/app/services/webhooks.service";
+import { BaseComponent } from "src/app/common/components/base.component";
+import { takeUntilDestroyed } from "@angular/core/rxjs-interop";
+import {
+    CreateWebhookSubscriptionSucces,
+    SubscribedEventType,
+    WebhookSubscriptionModalAction,
+} from "./create-subscription-modal.model";
+import { MaybeUndefined } from "src/app/interface/app.types";
+
+@Component({
+    selector: "app-create-subscription-modal",
+    templateUrl: "./create-subscription-modal.component.html",
+    changeDetection: ChangeDetectionStrategy.OnPush,
+})
+export class CreateSubscriptionModalComponent extends BaseComponent implements OnInit {
+    @Input({ required: true }) public datasetBasics: DatasetBasicsFragment;
+    @Input() public subscriptionData: CreateWebhookSubscriptionSucces;
+
+    public DROPDOWN_LIST: SubscribedEventType[] = [];
+    public secret: MaybeUndefined<string>;
+    public readonly WebhookSubscriptionStatus: typeof WebhookSubscriptionStatus = WebhookSubscriptionStatus;
+    public readonly WebhookSubscriptionModalAction: typeof WebhookSubscriptionModalAction =
+        WebhookSubscriptionModalAction;
+
+    public activeModal = inject(NgbActiveModal);
+    private fb = inject(FormBuilder);
+    private webhooksService = inject(WebhooksService);
+
+    public ngOnInit(): void {
+        this.webhooksService
+            .eventTypes()
+            .pipe(takeUntilDestroyed(this.destroyRef))
+            .subscribe((eventTypes: string[]) => {
+                this.DROPDOWN_LIST = eventTypes.map((eventType) => {
+                    return { name: this.eventTypesMapper(eventType), value: eventType };
+                });
+                if (this.subscriptionData) {
+                    this.createSubscriptionForm.patchValue({
+                        targetUrl: this.subscriptionData.input.targetUrl,
+                        eventTypes: this.subscriptionData.input.eventTypes,
+                        label: this.subscriptionData.input.label,
+                    });
+                    this.secret = this.subscriptionData?.secret;
+                }
+            });
+    }
+
+    public get targetUrlControl(): AbstractControl {
+        return this.createSubscriptionForm.controls.targetUrl;
+    }
+
+    public get eventTypesControl(): AbstractControl {
+        return this.createSubscriptionForm.controls.eventTypes;
+    }
+
+    public createSubscriptionForm = this.fb.nonNullable.group({
+        targetUrl: this.fb.control("", [Validators.required, Validators.pattern(AppValues.URL_PATTERN_ONLY_HTTPS)]),
```

Comment:
Perhaps we need to have an API extension so that you could test if the URL is fine not only by the format, but also that it's not a localhost address. This isn't something you could easily detect by text comparison, at the backend I'm resolving the IP address from the URL, and check whether it's a "loopback".

Reply:
We may need to discuss this